### PR TITLE
Add dockerfile

### PR DESCRIPTION
**Pull request by kaiehrhardt**
add Dockerfile to be able to use it in ci jobs running in a container environment

### CODE REVIEW

**Comment by preynolds-cs** [comment]
Hi @kaiehrhardt , Could you kindly include a README and the instructions for how to use this dockerfile to run a scrape?

**Comment by kaiehrhardt** [comment]
@preynolds-cs done

I have one more request. Is it possbile to add the image to a public sonatype community registry?
We could maybe reuse my github dockerhub ci: https://github.com/kaiehrhardt/iq-config-as-code/blob/master/.github/workflows/main.yml

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+ARG WORKDIR=/data
+ARG APPDIR=/iqcasc
+ARG USER=iqcasc
+ARG USER_ID=1000
+ARG GROUP=iqcasc
+ARG GROUP_ID=1000
+
+ENV PATH=/usr/local/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:${APPDIR}/.local/bin
+
+WORKDIR ${WORKDIR}
```
Comment (CSMonkee) [inline Dockerfile:13]: The variable WORKDIR is not used. Please remove it.

File: README.md [hunk 2]
```diff
@@ -51,13 +51,35 @@ Usage
...
+
+        $ docker build -t iq-server-casc:latest --build-arg BASEIMAGE=python:3.9.6-alpine3.14 .
+
+    In case you need some custom user ids or group ids, you could just build the image with those args:
+
+        $ docker build -t iq-server-casc:latest --build-arg USER_ID=1001 --build-arg GROUP_ID=1001 .
+
+    Run with local image:
+
+        docker run --rm -i -v $PWD:/data iq-server-casc:latest <iq-script>
```
Comment (CSMonkee) [inline README.md:70]: Please use: -v $PWD:/iqcasc

File: README.md [hunk 3]
```diff
@@ -51,13 +51,35 @@ Usage
...
+
+    Run with the latest upstream image:
+
+        docker run --rm -i -v $PWD:/data kehrhardt/iq-server-casc:latest <iq-script>
```
Comment (CSMonkee) [inline README.md:74]: Please use: -v $PWD:/iqcasc

File: README.md [hunk 4]
```diff
@@ -51,13 +51,35 @@ Usage
...
+
 
 Usage: iq-apply-config [ARGS]...
 
   Example usage:
 
     # Run python script though docker container with all packages installed on it!
-    docker run -w /tmp --rm -i -v $PWD:/tmp broadinstitute/python-requests /tmp/iq-apply-conf.py -f /tmp/conf/<conf-file>.json -u http://<iq-hostname>:<iq-port> -a <user>:<password>
+    docker run --rm -i -v $PWD:/data kehrhardt/iq-server-casc:latest iq-apply-conf -f /data/conf/<conf-file>.json -u http://<iq-hostname>:<iq-port> -a <user>:<password>
```
Comment (CSMonkee) [inline README.md:82]: Please use: -v $PWD:/iqcasc

File: README.md [hunk 5]
```diff
@@ -118,7 +140,7 @@ Usage: iq-scrape-config [ARGS]...
   Example usage:
     
     # Run python script though docker container with all packages installed on it!
-    docker run -w /tmp --rm -i -v $PWD:/tmp broadinstitute/python-requests iq-scrape-conf.py -u "http://<iq-hostname>:<iq-port>"
+    docker run --rm -i -v $PWD:/data kehrhardt/iq-server-casc:latest iq-scrape-conf -u "http://<iq-hostname>:<iq-port>" -a <user>:<password>
```
Comment (CSMonkee) [inline README.md:143]: please append to the arguments passed:  '-o /data'

File: README.md (hunk 4 above)
Comment (CSMonkee) [inline README.md:82]: Add comment to denote /data/conf/<config-file>.json is a unit test file.
Use /data/<config-file>.json to apply 'scraped' data.

File: README.md [hunk 6]
```diff
@@ -82,7 +104,7 @@ Usage: iq-healthcheck [ARGS]...
   Example usage:
     
     # Run python script though docker container with all packages installed on it!
-    docker run -w /tmp --rm -i -v $PWD:/tmp broadinstitute/python-requests iq-scrape-conf.py -u "http://<iq-hostname>:<iq-port>"
+    docker run --rm -i -v $PWD:/data kehrhardt/iq-server-casc:latest iq-healthcheck -u "http://<iq-hostname>:<iq-port>"
```
Comment (CSMonkee) [inline README.md:107]: Please add the following to the arguments passed: -t /data/healthcheck/templates/<template-config>.json

**Comment by CSMonkee** [comment]
As a docker newbie, I've enjoyed learning enough to understand why your changes work. Please make the slight changes commented and then we are good to merge this back. Thanks for your contribution and patience.

**Review by CSMonkee — CHANGES_REQUESTED** [review]
Please effect changes identified comments.

**Comment by kaiehrhardt** [comment]
> 
> 
> Please effect changes identified comments.

Done.



> 
> 
> @preynolds-cs done
> 
> I have one more request. Is it possbile to add the image to a public sonatype community registry? We could maybe reuse my github dockerhub ci: https://github.com/kaiehrhardt/iq-config-as-code/blob/master/.github/workflows/main.yml

How about that topic?

**Pull request merged**